code/_globalvars/lists/mobs.dm

```diff
@@ -34,3 +34,22 @@ GLOBAL_LIST_EMPTY(all_languages)
 GLOBAL_LIST_EMPTY(sentient_disease_instances)
 
 GLOBAL_LIST_EMPTY(latejoin_ai_cores)
+
+GLOBAL_LIST_EMPTY(mob_config_movespeed_type_lookup)
+
+/proc/initialize_mob_movespeed_type_lookup()
+	var/list/mob_typecache = typecacheof(/mob)
+	for(var/i in mob_typecache)
+		mob_typecache[i] = null

```

Comment:
I called it a typecache but you're not using it like a normal one, just create the list from a typesof call